# Contributor: ecoffey@gmail.com
# Mantainer: Imanol Celaya <ornitorrincos@archlinux-es.org>

pkgname=panda3d
pkgver=1.6.2
pkgrel=3
pkgdesc="3D engine with Python bindings"
url="http://www.panda3d.org"
depends=('python' 'freetype2' 'mesa' 'libtiff' 'openssl' 'libpng' 'gtk2' 'libjpeg')

source=(http://www.panda3d.org/download/panda3d-$pkgver/panda3d-$pkgver-linux32.tar.gz
panda3d.profile panda3d.pth panda3d.install)
[ "${CARCH}" = 'x86_64' ] && source[0]="http://www.panda3d.org/download/panda3d-${pkgver}/panda3d-${pkgver}-linux64.tar.gz"
arch=('i686' 'x86_64')
license=('BSD')
md5sums=('5ee2a850f4ef645e374858eb7bf6023d'
         '17c522241e60779aa8e50d532bb94e13'
         'df32d7c732b5cc8271e733a0d749c75f'
         '102365d5baac88c9a3b359f1a001fbe4')
[ "${CARCH}" = 'x86_64' ] && md5sums[0]="8c1f011e684ef69179b570f5892b46a6"

# Funny Quantum physics provoke panda
# not to build on some directories
# the error is:
#
# built/models/zup-axis.egg.pz
# built/bin/genpycode
# Cannot locate the root of the panda tree
#
# If it's the case build again in another directory
# /var/abs/local is known to host a black hole, so it works.
# Expected to be fixed in 1.7.0


build() {
  mkdir -p $pkgdir/opt/panda3d
  cd $srcdir/$pkgname-$pkgver

  makepanda/makepanda.py --everything --threads 2 || return 1

  cp built/* $pkgdir/opt/panda3d -r
  cp models $pkgdir/opt/panda3d -r
  cp direct/src $pkgdir/opt/panda3d/direct -r

  install -D $srcdir/panda3d.pth $pkgdir/usr/lib/python2.6/site-packages/panda3d.pth

  install -D -m755 $srcdir/panda3d.profile $pkgdir/etc/profile.d/panda3d.sh
  install -D -m755 $srcdir/$pkgname-$pkgver/doc/LICENSE $pkgdir/usr/share/licenses/$pkgname/LICENSE
}
